internal/transport/http2_server.go

```diff
@@ -540,10 +540,12 @@ func (t *http2Server) operateHeaders(ctx context.Context, frame *http2.MetaHeade
 	if len(mdata) > 0 {
 		s.ctx = metadata.NewIncomingContext(s.ctx, mdata)
 		if statsTags := mdata["grpc-tags-bin"]; len(statsTags) > 0 {
-			s.ctx = stats.SetIncomingTags(s.ctx, []byte(statsTags[len(statsTags)-1]))
+			mdata.Set("grpc-tags-bin", string([]byte(statsTags[len(statsTags)-1])))
+			s.ctx = metadata.NewIncomingContext(s.ctx, mdata)
 		}
 		if statsTrace := mdata["grpc-trace-bin"]; len(statsTrace) > 0 {
-			s.ctx = stats.SetIncomingTrace(s.ctx, []byte(statsTrace[len(statsTrace)-1]))
+			mdata.Set("grpc-trace-bin", string([]byte(statsTrace[len(statsTrace)-1])))
+			s.ctx = metadata.NewIncomingContext(s.ctx, mdata)
 		}
```

Comment:
Similar to above, just delete this.

Reply:
yes, i had initially deleted it but end2end test `TestStatsTagsAndTrace` was designed to check exactly one value being in header list for tags and traces. Deleted this and updated the tests `TestStatsTagsAndTrace`